src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,94 @@
+            int supplyActionsTotal = 0;
+            int buyActionsTotal = 0;
+            String data = bufferedReader.readLine();
+            while (data != null) {
+                String[] dataArray = data.split(",");
+                if (dataArray.length != DATA_LENGTH) {
+                    throw new RuntimeException("Invalid data format");
+                } else if (dataArray[INDEX_DATA_OPERATION_TYPE].equals(SUPPLY_ACTION)) {
+                    supplyActionsTotal++;
+                } else if (dataArray[INDEX_DATA_OPERATION_TYPE].equals(BUY_ACTION)) {
+                    buyActionsTotal++;
+                }
+                data = bufferedReader.readLine();
+            }
+            try (BufferedReader bufferedReader1 = new BufferedReader(new FileReader(file))) {
+                int[] supplyArray = new int[supplyActionsTotal];
+                int[] buyArray = new int[buyActionsTotal];
+                int supplyCount = 0;
+                int buyCount = 0;
+                data = bufferedReader1.readLine();
+                while (data != null) {
+                    String[] dataArray = data.split(",");
+                    if (dataArray[INDEX_DATA_OPERATION_TYPE].equals(SUPPLY_ACTION)) {
+                        supplyArray[supplyCount] = Integer.parseInt(dataArray[INDEX_AMOUNT]);
+                        supplyCount++;
+                    } else if (dataArray[INDEX_DATA_OPERATION_TYPE].equals(BUY_ACTION)) {
+                        buyArray[buyCount] = Integer.parseInt(dataArray[INDEX_AMOUNT]);
+                        buyCount++;
+                    }
+                    data = bufferedReader1.readLine();
+                }
+                int supplyTotal = 0;
+                int buyTotal = 0;
+                for (int buy: buyArray) {
+                    buyTotal += buy;
+                }
+                for (int supply: supplyArray) {
+                    supplyTotal += supply;
+                }
+                int[] result = {supplyTotal, buyTotal};
+                return result;
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File " + fromFileName + " not found", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from " + fromFileName, e);
+        }
+    }
 
+    private void writeDataToFile(int[] totalData, String toFileName) {
+        File file = new File(toFileName);
+        int supplySum = totalData[SUPPLY_INDEX_IN_ARRAY];
+        int buySum = totalData[BUY_INDEX_IN_ARRAY];
+        int result = supplySum - buySum;
+        StringBuilder report = new StringBuilder();
+        report.append(SUPPLY_ACTION).append(',').append(supplySum)
+            .append(System.lineSeparator());
+        report.append(BUY_ACTION).append(',').append(buySum).append(System.lineSeparator());
+        report.append(RESULT_ACTION).append(',').append(result);
```

Comment:
let's move this logic to a separate method `String createReport(int[] totalData)`